llvm-hs/src/LLVM/Internal/Operand.hs

```diff
@@ -45,6 +45,71 @@ import qualified LLVM.AST.Operand as A
 
 import LLVM.Internal.FFI.LLVMCTypes (mdSubclassIdP)
 
+genCodingInstance [t|A.DebugNameTableKind|] ''FFI.DebugNameTableKind
```

Comment:
Why did you have to move those?

Reply:
Sorry, I thought I explained that in details in the commits details.

Since GHC 9.2, template haskell code works as a "barrier" for visibility of instances. So an instances defined AFTER a TH piece of code cannot be seen before.

As an example, consider the following:

```haskell
data Foo = Foo

-- This functions uses the Eq instance for Foo
-- Which is define after someTH, so not visible here.
cmpFoo :: Foo -> Foo -> Bolo
cmpFoo a b = a == b

$(someTH)

deriving instance Eq Foo
```